src/core/database/query/lists.ts

```diff
@@ -14,7 +14,7 @@ export function calculateTotalAndPaginateList(
     // .with(['node', 'total']) TODO needed?
     .call(sort, sortInput, order)
     .with([
-      `collect(node.id)[${(page - 1) * count}..${page * count}] as items`,
+      `collect(distinct node.id)[${(page - 1) * count}..${page * count}] as items`,
```

Comment:
Haven't the nodes already been distinct-ed on line 12?

Reply:
@CarsonF  Yes, but results still contain duplicates.
So I added last distinct operator

Reply:
@CarsonF 
Do you agree with my solution?

Reply:
Ok it looks like since the `sort` callback can do more matches this is needed and does make a difference. 👍